LunrCoreLmdb/LmdbIndex.cs

```diff
@@ -0,0 +1,357 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using LightningDB;
+using Lunr;
+
+namespace LunrCoreLmdb
+{
+    public sealed class LmdbIndex : IReadOnlyIndex, IDisposable
+    {
+        private readonly LightningEnvironment _env;
+
+        public LmdbIndex(string path)
+        {
+            var config = new EnvironmentConfiguration
+            {
+                MaxDatabases = DefaultMaxDatabases,
+                MaxReaders = DefaultMaxReaders,
+                MapSize = DefaultMapSize
+            };
+            _env = new LightningEnvironment(path, config);
+            _env.Open();
+            CreateDatabaseIfNotExists(_env);
+        }
+
+        /// <summary>
+        /// A convenience function for configuring and constructing
+        /// a new lunr DelegatedIndex.
+        ///
+        /// An `LmdbBuilder` instance is created and the pipeline setup
+        /// with a trimmer, stop word filter and stemmer.
+        ///
+        /// This builder object is yielded to the configuration function
+        /// that is passed as a parameter, allowing the list of fields
+        /// and other builder parameters to be customized.
+        ///
+        /// All documents _must_ be added within the passed config function.
+        /// </summary>
+        /// <example>
+        /// var idx = Index.Build(async builder =>
+        /// {
+        ///      builder
+        ///         .AddField("title")
+        ///         .AddField("body");
+        ///
+        ///      builder.ReferenceField = "id";
+        ///
+        ///      foreach(Document doc in documents)
+        ///      {
+        ///          builder.add(doc);
+        ///      }
+        /// });
+        /// </example>
+        /// <param name="stopWordFilter">An optional stopword filter. Default is English.</param>
+        /// <param name="stemmer">An optional stemmer. Default is English.</param>
+        /// <param name="config">A Configuration function.</param>
```

Comment:
This is something I'm fixing in the core library as well, we have a bunch of missing doc comments for optional parameters here.

Reply:
I added the missing params.